test/tabbable.test.js

```diff
@@ -40,117 +58,125 @@ describe('tabbable', function() {
+          'inputTabindex0',
+          'spanTabindex0',
+          'dimensionlessParent',
+          'dimensionlessParent-dimensionless',
+        ];
+        assert.deepEqual(actual, expected);
+      });
 
-  it('changing content', function() {
-    var loadedFixture = fixture('changing-content');
-    var actualA = loadedFixture.getTabbableIds();
-    var expectedA = [
-      'visible-button-1',
-      'visible-button-2',
-      'visible-button-3',
-    ];
-    assert.deepEqual(actualA, expectedA);
-
-    document.getElementById('initially-hidden').style.display = 'block';
-
-    var actualB = loadedFixture.getTabbableIds();
-    var expectedB = [
-      'visible-button-1',
-      'visible-button-2',
-      'visible-button-3',
-      'initially-hidden-button-1',
-      'initially-hidden-button-2',
-    ];
-    assert.deepEqual(actualB, expectedB);
+      it('non-linear', function() {
+        var originalSort = Array.prototype.sort;
+
+        // This sort piggy-backs on the default Array.prototype.sort, but always
+        // orders elements that were compared to be equal in reverse order of their
+        // index in the original array. We do this to simulate browsers who do not
+        // use a stable sort algorithm in their implementation.
+        Array.prototype.sort = function(compareFunction) {
+          return originalSort.call(this, function(a, b) {
+            var comparison = compareFunction ? compareFunction(a, b) : a - b;
+            return comparison || this.indexOf(b) - this.indexOf(a);
+          })
+        };
+        var actual = fixture('non-linear').getTabbableIds();
+        Array.prototype.sort = originalSort;
+        var expected = [
+          // 1
+          'input-1',
+          'href-anchor-1',
+          // 2
+          'button-2',
+          // 3
+          'select-3',
+          'tabindex-div-3',
+          // 4
+          'tabindex-hrefless-anchor-4',
+          //12
+          'textarea-12',
+          // 0
+          'input',
+          'select',
+          'href-anchor',
+          'textarea',
+          'button',
+          'tabindex-div-0',
+        ];
+        assert.deepEqual(actual, expected);
+      });
+
+      it('changing content', function() {
+        var loadedFixture = fixture('changing-content');
+        var actualA = loadedFixture.getTabbableIds();
+        var expectedA = [
+          'visible-button-1',
+          'visible-button-2',
+          'visible-button-3',
+        ];
+        assert.deepEqual(actualA, expectedA);
+
+        loadedFixture.getDocument().getElementById('initially-hidden').style.display = 'block';
+
+        var actualB = loadedFixture.getTabbableIds();
+        var expectedB = [
+          'visible-button-1',
+          'visible-button-2',
+          'visible-button-3',
+          'initially-hidden-button-1',
+          'initially-hidden-button-2',
+        ];
+        assert.deepEqual(actualB, expectedB);
+      });
+    });
   });
 });
```

Comment:
As part of the PR, can we add an entry to the changelog and some content to the documentation?

Reply:
Sure thing, I'll add something to `CHANGELOG.md` for version `1.0.8` and describe the iframe support addition—did you have any additional documentation targets in mind?